spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/session/SessionAutoConfigurationJdbcTests.java

```diff
@@ -86,4 +88,15 @@ public void customTableName() {
 				.queryForList("select * from FOO_BAR")).isEmpty();
 	}
 
+	@Test
+	public void customTableNameWithDefaultSchemaFails() {
+		this.thrown.expect(BeanCreationException.class);
+		this.thrown.expectMessage(containsString(
+				"Database initializer cannot be configured"));
+		load(Arrays.asList(EmbeddedDataSourceConfiguration.class,
+				DataSourceTransactionManagerAutoConfiguration.class),
+				"spring.session.store-type=jdbc",
+				"spring.session.jdbc.table-name=FOO_BAR");
```

Comment:
If we know it's not going to work then I'd throw an exception as early as possible rather than logging the warning. If we can't be sure if it'll work or not then the warning seems potentially misleading and we should just continue and let it throw an exception later on if it turns out that it doesn't work.